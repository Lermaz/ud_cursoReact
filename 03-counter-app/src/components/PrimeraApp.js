import { Fragment } from "react";
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {


  return (
    <Fragment>
      <h1>{ saludo }</h1>
      <p>{ subtitulo }</p>
    </Fragment>
  );
}

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;