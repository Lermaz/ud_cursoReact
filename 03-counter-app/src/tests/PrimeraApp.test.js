import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import PrimeraApp from "../components/PrimeraApp";

describe('Pruebas en PrimeraApp', () => {
  
  /*test('Debe de mostrar el mensaje "Hola, soy Isaac"', () => {
    
    const saludo = 'Hola, soy Isaac';
    const wrapped = render( <PrimeraApp saludo={ saludo } /> );

    expect( wrapped.getByText(saludo) ).toBeInTheDocument();
  });*/

  test('Debe de mostrar <PrimeraApp /> correctamente', () => {
    
    const saludo = 'Hola, soy Isaac';
    const wrapped = shallow( <PrimeraApp saludo={ saludo } /> );

    expect( wrapped ).toMatchSnapshot();
  });

  test('Debe de mostrar el subtitulo enviado por props', () => {
    
    const saludo = 'Hola, soy Isaac';
    const subtitulo = 'Soy un subtitulo';

    const wrapped = shallow( 
      <PrimeraApp 
        saludo={ saludo }
        subtitulo= { subtitulo } /> 
    );

    const textoParrafo = wrapped.find('p').text();
    
    expect( textoParrafo ).toBe( subtitulo );
  });
})