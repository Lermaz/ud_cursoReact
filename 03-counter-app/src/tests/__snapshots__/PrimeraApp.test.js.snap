// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Isaac
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
